From 672e71ba4c9c1372cff420840e70cec699062841 Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Mon, 17 Oct 2016 11:59:35 +0200
Subject: [PATCH 18/24] STM: waylandsink: support dmabuf YUY2 and I420 pixel
 formats

Useful for webcam zero-copy.
Those pixel formats will be supported by default in the waylandsink
patch "wayland: add DMABuf support to wayland sink":
https://bugzilla.gnome.org/show_bug.cgi?id=711155

Change-Id: Ia72dc8a5d4ba396ad5f15b9c492073b4141ba7b3
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
Reviewed-on: https://gerrit.st.com/59169
Reviewed-by: Fabien DESSENNE <fabien.dessenne@st.com>
Tested-by: Fabien DESSENNE <fabien.dessenne@st.com>
---
 ext/wayland/wlvideoformat.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/wayland/wlvideoformat.c b/ext/wayland/wlvideoformat.c
index 9648b7c..e6a5664 100644
--- a/ext/wayland/wlvideoformat.c
+++ b/ext/wayland/wlvideoformat.c
@@ -111,6 +111,8 @@ static const wl_DmabufVideoFormat dmabuf_formats[] = {
 #endif
   {DRM_FORMAT_RGB565, GST_VIDEO_FORMAT_RGB16},
   {DRM_FORMAT_NV12, GST_VIDEO_FORMAT_NV12},
+  {DRM_FORMAT_YUYV, GST_VIDEO_FORMAT_YUY2},
+  {DRM_FORMAT_YUV420, GST_VIDEO_FORMAT_I420},
 };
 
 enum wl_shm_format
-- 
2.7.4

